### PR TITLE
Add loading state when fetching swap price details

### PR DESCRIPTION
**Pull request by TheJarX**
This PR adds a loading gif when fetching price details in the Swap page. 

Resolves: https://github.com/tallycash/extension/issues/2629

https://user-images.githubusercontent.com/35399391/205707966-d6182444-4273-4f9c-9d44-54e8a1f2e12d.mov

## To test
- [ ] Verify that the gif appears when fetching price details

Latest build: [extension-builds-2723](https://github.com/tallycash/extension/suites/9720185917/artifacts/466178801) (as of Wed, 07 Dec 2022 12:07:50 GMT).

### CODE REVIEW

**Comment by jagodarybacka** [comment]
Hey @TheJarX we recently added component [SharedLoadingDoggo](https://github.com/tallycash/extension/blob/main/ui/components/Shared/SharedLoadingDoggo.tsx), could you please use it here?

**Comment by TheJarX** [comment]
Looks better, thanks for the heads up @jagodarybacka!

https://user-images.githubusercontent.com/35399391/206158492-ff84e461-5677-446a-bdaf-4e6bcd7532f7.mov

File: ui/pages/Swap.tsx [hunk 1]
```diff
@@ -635,6 +636,16 @@ export default function Swap(): ReactElement {
                 }}
                 label={t("swap.to")}
               />
+              {priceDetails === undefined &&
+              sellAmount &&
```
Comment (jagodarybacka) [inline ui/pages/Swap.tsx:640]: Let's do it like this so it goes both ways 😄 
```suggestion
              (sellAmount || buyAmount) &&
```

**Comment by TheJarX** [comment]
Oh yeah nice catch, btw I was checking the scenario when it says '`Insufficient liquidity for this trade`

https://user-images.githubusercontent.com/35399391/206170791-b7d68cd6-0b1e-46f1-b7e9-ef14a8dac82e.mov

**Review by jagodarybacka — APPROVED** [review]
Hmm can we easily check if the insufficient liquidity error was thrown? 
If we can't I think it is good enough. I'll let @VladUXUI take a look from UX perspective before merging.

**Comment by VladUXUI** [comment]
Thanks @TheJarX !
This is good! and i'm ok to leave the loader in for insufficient balance if it's not an easy fix. But we should rectify that shortly

**Review by VladUXUI — APPROVED** [review]

**Pull request merged**